test/common.py

```diff
@@ -899,20 +901,15 @@ def setUp(self):
       else:
         print('Creating new test output directory')
         ensure_dir(self.working_dir)
-    else:
-      self.working_dir = tempfile.mkdtemp(prefix='emscripten_test_' + self.__class__.__name__ + '_', dir=self.temp_dir)
     os.chdir(self.working_dir)
 
-    if not EMTEST_SAVE_DIR:
-      self.has_prev_ll = False
-      for temp_file in os.listdir(shared.TEMP_DIR):
-        if temp_file.endswith('.ll'):
-          self.has_prev_ll = True
+  def runningInParallel(self):
+    return getattr(self, 'is_parallel', False)
 
   def tearDown(self):
-    if not EMTEST_SAVE_DIR:
+    os.chdir(os.path.dirname(self.get_dir()))
```

Comment:
Why move this code up? Is it needed even if we don't enter the `if` below?

The comment that used to justify it (about rmtree) should perhaps move with it, if we must move it, but that comment referred to the combination of old likes 915-916 so separate comments might be needed.

Reply:
No good reason I guess.  Will see if I can move it back.

Reply:
Done